gensim/models/word2vec.py

```diff
@@ -804,6 +807,23 @@ def update_weights(self):
         # do not suppress learning for already learned words
         self.wv.vectors_lockf = np.ones(1, dtype=REAL)  # 0.0 values suppress word-backprop-updates; 1.0 allows
 
+    def init_sims(self, replace=False):
```

Comment:
This method is obsolete & shouldn't exist; code that thinks it needs it (1) probably doesn't anymore; (2) should do more explicitly & specifically what it needs elsewhere or more directly, as for example on KeyedVectors, perhaps using the `unit_normalize_all()` method. 

If it's retained for backward compatibility, it should be marked deprecated & just pass-through to the similarly deprecated method on KeyedVectors, without having odd side-effects like the deletion of `syn1` here (which inexplicably has no matching deletion of `syn1neg` for negative-sampling models?). If some caller really needs those layers deleted, it'd be better for the caller to do that explicitly.

Reply:
Re: the question in comments - IMO `init_sims()` also shouldn't exist on `Doc2Vec` & `FastText`; its persistence in those places is just incomplete removal of obsolete stuff. Keeping it on `KeyedVectors`, with both a deprecation warning & louder warning if used with `replace=True`, is just a guidepost to help those dealing with its more-general obsolescence.

Reply:
Alright, I'll remove it. Let's keep this in mind for the migration guide.
Can you craft a more noob-friendly error message for KeyedVectors?

Reply:
I don't think noobs will have been likely to accessing `.vocab` directly, unless it's in cut & pasted code. And other than the existing pointers to the names of new properties/methods offering equivalent functionality, I don't have ideas for more noob-friendly verbiage to put in the actual error text. (I don't think they'll care about the rationales, and I don't have enough insight into why they might be calling it to include any mini-tutorials on how to update their code.)

Reply:
Hmm. I hoped it might be something as simple as `model.init_sims() no longer exists; call model.kw.init_sims() instead`? 
What's the complication?

Reply:
Ah, for some reason I got this thread confused with the `.vocab` discussion, & didn't see a path to any message better than the (already-4-lines-of-text) message accompanying the `NotImplementedError` there. Yes, if you're OK with a hard-fail on uses of `init_sims()`, such a message makes sense. And it could be on `KeyedVectors`, too, pointing users of any variant of that method to the info that (1) they probably no longer need to do init_sims; (2) single normed vectors are easy to get via `get_vector()`; (3) if they want to get a big array of all unit-normed vectors, some (to be added distinctively-named) method is available; (4) if they want to ensure norms are fresh after performing atypical out-of-band vector tampering,  call `fill_norms()`.

Reply:
Yes, that's exactly it.

I thought we'd put that info directly into the error message, but perhaps a link to some permanent URL (Github wiki?) gives us more flexibility, at the cost of decreased user convenience, for very chatty errors.